src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_eckey.c

```diff
@@ -0,0 +1,132 @@
+            free(r);
+        }
+    }
+}
+
+EC_KEY* CryptoNative_EcKeyCreateByOid(const char* oid)
+{
+    JNIEnv* env = GetJNIEnv();
+
+    jstring oidStr = JSTRING(oid);
+    jstring ec = JSTRING("EC");
+
+    // First, generate the key pair based on the curve defined by the oid.
+    jobject paramSpec = (*env)->NewObject(env, g_ECGenParameterSpecClass, g_ECGenParameterSpecCtor, oidStr);
+    ReleaseLRef(env, oidStr);
+    
+    jobject keyPairGenerator = (*env)->CallStaticObjectMethod(env, g_keyPairGenClass, g_keyPairGenGetInstanceMethod, ec);
+    (*env)->CallVoidMethod(env, keyPairGenerator, g_keyPairGenInitializeMethod, paramSpec);
+
+    ReleaseLRef(env, paramSpec);
+    if (CheckJNIExceptions(env))
+    {
+        ReleaseLRef(env, ec);
+        ReleaseLRef(env, keyPairGenerator);
+        return NULL;
+    }
+
+    jobject keyPair = (*env)->CallObjectMethod(env, keyPairGenerator, g_keyPairGenGenKeyPairMethod);
+    
+    ReleaseLRef(env, keyPairGenerator);
+    
+    // Now that we have the key pair, we can get the curve parameters from the public key.
+    jobject keyFactory = (*env)->CallStaticObjectMethod(env, g_KeyFactoryClass, g_KeyFactoryGetInstanceMethod, ec);
+    jobject publicKey = (*env)->CallObjectMethod(env, keyPair, g_keyPairGetPublicMethod);
+    jobject keySpec = (*env)->CallObjectMethod(env, keyFactory, g_KeyFactoryGetKeySpecMethod, publicKey, g_ECPublicKeySpecClass);
+
+    ReleaseLRef(env, ec);
+    ReleaseLRef(env, publicKey);
+    ReleaseLRef(env, keyFactory);
+
+    if (CheckJNIExceptions(env))
+    {
+        ReleaseLRef(env, keySpec);
+        ReleaseLRef(env, keyPair);
+        return NULL;
+    }
+
+    jobject curveParameters = (*env)->CallObjectMethod(env, keySpec, g_ECPublicKeySpecGetParams);
+
+    return CryptoNative_NewEcKey(ToGRef(env, curveParameters), ToGRef(env, keyPair));
+}
+
+int32_t CryptoNative_EcKeyGenerateKey(EC_KEY* eckey)
+{
+    // We have to generate the key in CryptoNative_EcKeyCreateByOid to get the curve parameters,
+    // so by the time we get here, we've already generated the key.
+    return SUCCESS;
+}
+
+int32_t CryptoNative_EcKeyUpRef(EC_KEY* r)
+{
+    if (!r)
+        return FAIL;
+    r->refCount++;
+    return SUCCESS;
+}
+
+int32_t CryptoNative_EcKeyGetSize(const EC_KEY* key, int32_t* keySize)
+{
+    if (!keySize)
+        return 0;
+    
+    *keySize = 0;
+
+    if (!key)
+        return 0;
+
+    JNIEnv* env = GetJNIEnv();
+
+    jobject curve = (*env)->CallObjectMethod(env, key->curveParameters, g_ECParameterSpecGetCurve);
+    jobject field = (*env)->CallObjectMethod(env, curve, g_EllipticCurveGetField);
+    *keySize = (*env)->CallIntMethod(env, field, g_ECFieldGetFieldSize);
+
+    ReleaseLRef(env, field);
+    ReleaseLRef(env, curve);
+    return 1;
+}
+
+int32_t CryptoNative_EcKeyGetCurveName2(const EC_KEY* key, int32_t* nidName)
+{
+    // TODO: The public APIs do not support getting a name from an ECKeyParameters object.
+    return FAIL;
```

Comment:
Why not block this in managed code?

Reply:
One of my goals with this PR was to not change the managed surface area if possible. I was just trying to fill in the shim.

It seems like that might not be possible now, so I'll fold this into the custom managed code for Android.

Reply:
I've made this safely no-op in the shim for now since the reason to make lots of managed code changes in this PR (using the P1363 encoding for the DSA signatures instead of DER encoding) isn't happening since Android doesn't support the P1363 formats.